### ISSUE TITLE
Behaviour of crossfader after disabling AutoDJ -> Disabling crossfader?

### ISSUE THREAD

**Issue by Eve00000** [mixxxdj/mixxx#12985]
### Bug Description

AutoDJ uses the crossfader to pass the audio signal. When AutoDJ is enabled the crossfader starts to move. When AutoDJ is disabled the crossfader stays 'in position".
If internal mixer 'and crossfader are hidden in settings (working with external mixer)  crossfader must be reset to the center (settings - display mixer .....)
Correct me if I'm wrong but there is no way to disable the crossfader (and internal mixer)

### Version

2.5

### OS

Win10 / Win 11

**Comment by ronso0** [comment]
> AutoDJ uses the crossfader to pass the audio signal. When AutoDJ is enabled the crossfader starts to move. When AutoDJ is disabled the crossfader stays 'in position". When using AutoDJ as Backup for going to the bathroom this is not 'handy'
If internal mixer 'and crossfader are hidden in settings (working with external mixer, each deck a channel) crossfader must be reset to the center (settings - display mixer .....)
Correct me if I'm wrong but there is no way to disable the crossfader (and internal mixer)

That's how it is: AutoDJ works with the crossfader.
What exactly are you proposing?
(note there is #11571 and it's Pros and Cons have been discussed at lengths, just someone has to implement it)
(note there is #7520 **Volume slider mode for Auto DJ**)

**Comment by ronso0** [comment]
> When setting deck3 to leftside and deck4 to right side while deck 1 & 2 are centered AutoDJ starts using Deck 3 & 4 for a couple of seconds and then causes mixxx to crash.

Will try to reproduce.

**Comment by ronso0** [comment]
> When setting deck3 to leftside and deck4 to right side while deck 1 & 2 are centered AutoDJ starts using Deck 3 & 4 for a couple of seconds and then causes mixxx to crash.

**1** I tried this
* decks 1/2 --|--, 3 |----, 4 ----|
* start AutoDJ
* uses decks 3/4
* deck4 playing, deck3 stopped
* make deck1 |----
* click Fade Now
* :warning: "AutoDJ requires two decks at the opposite side of the crossfader", AutoDJ turns OFF

**2** and this
* deck3 |----, deck1 --|--, deck2 ----|, deck4 ----|
* start AutoDJ, plays deck3
* Fade Now
* fades to deck2
* uses decks 3 + 2 without issues
* while deck3 is paying: set deck --|-- (should switch to deck4 which is still ----| )
* :warning: "AutoDJ requires two decks at the opposite side of the crossfader", AutoDJ turns OFF

So there is definitely something to improve in the deck attributes logic.

**Comment by Eve00000** [comment]
[sorry for the music, kust testing at home](https://youtu.be/uJt5wX6YPXI)
The crossfader error was not that bad, I just missed a mix ;-)

**Comment by ronso0** [comment]
When you say "crash" you probably hit this debug assert
```
DEBUG ASSERT: "pFromDeck && pToDeck" in function void AutoDJProcessor::calculateTransition(DeckAttributes*, DeckAttributes*, bool) at ./src/library/autodj/autodjprocessor.cpp:1233
[1]    33203 abort (core dumped)  ./mixxx --developer --settingsPath /home/ronso/.mixxx_dev
```
Mixxx 2.5 is still alpha, i.e. like beta it is built with `DEBUG_ASSERTIONS_FATAL` set ON (unless you changed that setting, please verify).
So this is not a "crash" (entirely unhandled exception with Mixxx being unable to deal with it) but a helper that indicates that code path should be improved.

**Comment by Eve00000** [comment]
No it is another:
DEBUG ASSERT: "beatFraction > 0 && beatFraction < 1" in function bool __cdecl mixxx::Beats::findPrevNextBeats(class mixxx::audio::FramePos,class mixxx::audio::FramePos *,class mixxx::audio::FramePos *,bool) const at D:\mixxx-main\src\track\beats.cpp:415

I know it's alpha, but by using and testing I can help finding bugs. By using a release we're not getting further.

**Comment by ronso0** [comment]
Can you reproduce this hit?

**Comment by Eve00000** [comment]
Yes, the video is the 7th time or so.... I wrote it in my original post. How many times you want it reproduced? Don't let me make a video each time :-)

**Comment by ronso0** [comment]
> I know it's alpha, but by using and testing I can help finding bugs

Yep, thanks for that. I wanted to point out it's not crashing, in this case (release, FATAL Off) the function simply returns.

> How many times you want it reproduced?

Are wa talking about the same assert here? I meant: can you reporduce the assert hit in beats.cpp:415 with the described test procedure AutoDJ?

In general I'm mostly working mobile, so for saving bandwidth/volume I prefer some simple repro steps over a 5min FullHD video, which also way more convenient for anynone passing by ; )

**Comment by daschuer** [comment]
@Eve00000 the bug title is basically a request for the Volume slider mode of AUTO DJ https://github.com/mixxxdj/mixxx/issues/7520, a duplicate. 

In the discussion we have revealed the another requirement of disabling the internal mixer. 
Is this correct? Can we adjust the title accordingly?
